library/general/test/report_test.rb

```diff
@@ -121,4 +123,58 @@
       expect(subject.GetMessages(0, 1, 0, 0)).to match(/Message/)
     end
   end
+
+  describe ".Settings" do
+    DATA_PATH =  File.join(File.dirname(__FILE__), 'data')
+    let(:ay_profile) { File.join(DATA_PATH, 'ay_profile.yml') }
+    let(:default_normal) { File.join(DATA_PATH, 'default_normal_installation.yml') }
+    let(:default_ay) { File.join(DATA_PATH, 'default_ay_installation.yml') }
+    let(:result_ay) { File.join(DATA_PATH, 'ay_installation.yml') }
+
+    context "while normal installation" do
+      it "check default entries" do
+        Yast::Mode.SetMode("installation")
```

Comment:
Um, I'd avoid setting the mode directly, this changes the global state outside the tests. It might have some side effects...

Simply mocking `Mode.mode` should work I think.
```ruby
allow(Yast::Mode).to receive(:mode).and_return("installation")
```